### PR TITLE
SimpleContent support fix

### PR DESCRIPTION
**Pull request by fernandor777**
FIxes SimpleContent support.

### CODE REVIEW

**Comment by nmco** [comment]
Thx @fernandor777, so if I understood correctly this will convert, and only this case, lists of simple values to a lits of complex values right? We need to be careful here, the produced schema needs to be backwards compatible ☠️

**Comment by fernandor777** [comment]
This PR will transform only complexType with simpleContent elements to Complex values, there is no difference between 1 element or list.  Regular simpleType and complexType elements stay the same.

File: xsd2xml/src/main/java/it/geosolutions/xsd2gml/ComplexTypeConverter.java [hunk 1]
```diff
@@ -68,6 +84,21 @@ private void addGmlFeatureType(
         merger.addProperties(outputSchema, sequence);
     }
 
+    private void handleSimpleContent(Document outputSchema, Element outputSchemaRootNode, QName targetNamespace,
```
[the remark this answers is not in the document]
Comment (fernandor777) [reply xsd2xml/src/main/java/it/geosolutions/xsd2gml/ComplexTypeConverter.java]: Right, so we can do CollectionLink on Hale.

File: xsd2xml/src/main/java/it/geosolutions/xsd2gml/ComplexTypeConverter.java [hunk 2]
```diff
@@ -83,7 +114,7 @@ private void addGmlFeatureConcreteElement(
         // create the XML element for the concrete instance
         Element element = outputSchema.createElementNS(XML_NAMESPACE, "xs:element");
         element.setAttribute("name", getName(name, null));
-        element.setAttribute("type", getTypeName(name, targetNamespace));
+	element.setAttribute("type", /* isSimpleContent(typeDefinition) ? name : */getTypeName(name, targetNamespace));
```
[the remark this answers is not in the document]
Comment (fernandor777) [reply xsd2xml/src/main/java/it/geosolutions/xsd2gml/ComplexTypeConverter.java]: Fixed

File: xsd2xml/src/main/java/it/geosolutions/xsd2gml/SchemaWalker.java [hunk 3]
```diff
@@ -88,9 +88,9 @@ private void walkTypeProperties(String unqualifiedTypeName) {
             final Element simpleExtensionElement =
                     searchElement(complexType, "./simpleContent/extension");
             if (simpleExtensionElement != null && simpleExtensionElement.hasAttribute("base")) {
-                rootSimpleTypes.add(complexType);
+		// rootSimpleTypes.add(complexType);
                 walkTypeProperties(extractUnqualifiedTypeName(simpleExtensionElement, "base"));
-                return;
+		// return;
```
[the remark this answers is not in the document]
Comment (fernandor777) [reply xsd2xml/src/main/java/it/geosolutions/xsd2gml/SchemaWalker.java]: Fixed

File: xsd2xml/src/main/java/it/geosolutions/xsd2gml/ComplexTypeConverter.java [hunk 4]
```diff
@@ -33,7 +42,8 @@
      */
     void toGmlFeature(Document outputSchema, Element outputSchemaRootNode, QName targetNamespace) {
         addGmlFeatureConcreteElement(outputSchema, outputSchemaRootNode, targetNamespace);
-        addGmlFeatureType(outputSchema, outputSchemaRootNode, targetNamespace);
+	// if (!isSimpleContent(typeDefinition))
+	    addGmlFeatureType(outputSchema, outputSchemaRootNode, targetNamespace);
```
[the remark this answers is not in the document]
Comment (fernandor777) [reply xsd2xml/src/main/java/it/geosolutions/xsd2gml/ComplexTypeConverter.java]: Fixed

File: xsd2xml/src/main/java/it/geosolutions/xsd2gml/Xsd2Gml.java [hunk 5]
```diff
@@ -29,17 +29,13 @@ public Xsd2Gml(
...
         walker.getRootComplexTypes().entrySet().stream()
-                .filter(e -> !isSimpleContent(e.getKey()))
                 .forEach(
                         e -> {
                             ComplexTypeConverter converter =
                                     new ComplexTypeConverter(inputSchema, e.getKey(), e.getValue());
                             converter.toGmlFeature(outputSchema, outputRootNode, targetNamespace);
                         });
         walker.getRootSimpleTypes().forEach(simpleType -> adoptSimpleType(simpleType));
-        walker.getRootComplexTypes().entrySet().stream()
```
[the remark this answers is not in the document]
Comment (fernandor777) [reply xsd2xml/src/main/java/it/geosolutions/xsd2gml/Xsd2Gml.java]: In last fix we added a second stream for complexTypes with simpleContent, not needed anymore.

**Comment by fernandor777** [comment]
Thanks for your comments, all fixed

**Comment by nmco** [comment]
Thx @fernandor777, merged!

**Pull request merged**